client/blocks/taxonomy-manager/list.jsx

```diff
@@ -153,6 +160,11 @@ export class TaxonomyManagerList extends Component {
 		} );
 		const hasDefaultSetting = taxonomy === 'category';
 
+		// add a bogus query arg to trigger a new request on delete
+		if ( this.state.deletedItem ) {
+			query.deletedItem = this.state.deletedItem;
```

Comment:
I think this might solve the issue
```js
itemDeleted = () => {
    this.setState(
        { deletingItem: true },
        () => this.setState( { deletingItem: false } )
    );
}
```
followed by
```jsx
{  ! this.state.deletingItem && <QueryTerms
    key={ `query-${ page }` }
    siteId={ siteId }
    taxonomy={ taxonomy }
    query={ { ...query, page } } /> }
```

But even that feels wrong since we're tricking the `QueryTerms` component into refreshing. In my opinion I think the request should look intentional.